src/SqlPersistence/ScriptRunner.cs

```diff
@@ -29,30 +40,37 @@ public static class ScriptRunner
             Guard.AgainstNullAndEmpty(nameof(scriptDirectory), scriptDirectory);
             sqlDialect.ValidateTablePrefix(tablePrefix);
 
-            using (var connection = await connectionBuilder.OpenConnection().ConfigureAwait(false))
-            using (var transaction = connection.BeginTransaction())
+            if (shouldInstallOutbox)
             {
-                if (shouldInstallOutbox)
-                {
-                    await InstallOutbox(scriptDirectory, connection, transaction, tablePrefix, sqlDialect).ConfigureAwait(false);
-                }
+                await Execute<StorageType.Outbox>(InstallOutbox, scriptDirectory, tablePrefix, sqlDialect, connectionBuilder).ConfigureAwait(false);
+            }
 
-                if (shouldInstallSagas)
-                {
-                    await InstallSagas(scriptDirectory, connection, transaction, tablePrefix, sqlDialect).ConfigureAwait(false);
-                }
+            if (shouldInstallSagas)
+            {
+                await Execute<StorageType.Sagas>(InstallSagas, scriptDirectory, tablePrefix, sqlDialect, connectionBuilder).ConfigureAwait(false);
+            }
 
-                if (shouldInstallSubscriptions)
-                {
-                    await InstallSubscriptions(scriptDirectory, connection, transaction, tablePrefix, sqlDialect).ConfigureAwait(false);
-                }
+            if (shouldInstallSubscriptions)
+            {
+                await Execute<StorageType.Subscriptions>(InstallSubscriptions, scriptDirectory, tablePrefix, sqlDialect, connectionBuilder).ConfigureAwait(false);
+            }
 
-                if (shouldInstallTimeouts)
+            if (shouldInstallTimeouts)
+            {
+                await Execute<StorageType.Timeouts>(InstallTimeouts, scriptDirectory, tablePrefix, sqlDialect, connectionBuilder).ConfigureAwait(false);
+            }
+        }
+
+        static async Task Execute<T>(Func<string, DbConnection, DbTransaction, string, SqlDialect, Task> installAction, string scriptDirectory, string tablePrefix, SqlDialect sqlDialect, Func<Type, DbConnection> connectionBuilder)
+            where T : StorageType
+        {
+            using (var connection = connectionBuilder(typeof(T)))
+            {
+                await connection.OpenAsync().ConfigureAwait(false);
+                using (var transaction = connection.BeginTransaction())
                 {
-                    await InstallTimeouts(scriptDirectory, connection, transaction, tablePrefix, sqlDialect).ConfigureAwait(false);
+                    await installAction(scriptDirectory, connection, transaction, tablePrefix, sqlDialect).ConfigureAwait(false);
                 }
-
-                transaction.Commit();
```

Comment:
Should we worry that the installers won't run transactionally? Or, vs. the cost of trying to group things by connection builder and somehow execute common ones together in a transaction still?

Reply:
I think this is a bug? I don't see where those transactions are commited.

Reply:
I don't think have a single transaction is necessary here as these installers are not destructive.